### DISCUSSION TITLE
vcpkg why command

### DISCUSSION THREAD

**Discussion by silverqx** [microsoft/vcpkg#33931] [General]
Is there any command that can tell why some package is installed? I found the `depend-info` command only.

**Comment by autoantwort** [comment]
So a command that list all installed ports that depends on a given port?

**Comment by silverqx** [comment]
No, something like `vcpkg why libevent`  or `vcpkg why harfbuzz[icu]` it's practically the opposite of the `depend-info` command, but I didn't find anything like that.

**Comment by silverqx** [comment]
> So a command that list all installed ports that depends on a given port?

Yeah, you wrote it right, I didn't understand it at first. List all installed ports that depend on the given port, that is right.

**Comment by silverqx** [comment]
So nothing like that currently exists, right? I mean I can close this

**Reply by autoantwort** [reply]
Right. You can leave this open as a feature request. 

May I ask what the use case is?

**Comment by silverqx** [comment]
Yesterday happened to me this, I had installed the `qt5-base`, which has a lot of dependencies and I wanted to uninstall them and I didn't know which dependency packages I could uninstall to not corrupt the current vcpkg installation. The `depend-info` command helped a lot but the `why` command would be much better for this scenario.

**Reply by dg0yt** [reply]
There is no record of whether `tiff` was installed manually or as automatically a dependency of `qt5-base`.
We can only ask for dependencies but not for actual triggers.

**Comment by dg0yt** [comment]
For now, you can at least inspect the full cone of destruction (including transitive reverse dependencies) with
~~~
vcpkg remove --dry-run tiff
~~~

**Comment by silverqx** [comment]
> ```
> vcpkg remove --dry-run tiff
> ```

I didn't know this, I thought that `vcpkg` uninstalls the given package only but this is doing exactly what I'm expecting and is a very good alternative to the `why` command. Thank you for clarifying.

You practically can't destroy the current vcpkg installation because the `remove` command informs you that something depends on it which is great.

> There is no record of whether tiff was installed manually or as automatically

This is true, first I thought that it isn't but when I'm thinking about this then it is clear that it is important.

**Comment by dg0yt** [comment]
For finding direct reverse dependencies of an *installed* package, you can use the output of:
~~~
$ grep -l "^curl " installed/x64-linux/share/*/vcpkg_abi_info.txt
installed/x64-linux/share/cmake-user/vcpkg_abi_info.txt
installed/x64-linux/share/gdal/vcpkg_abi_info.txt
installed/x64-linux/share/netcdf-c/vcpkg_abi_info.txt
installed/x64-linux/share/proj/vcpkg_abi_info.txt
installed/x64-linux/share/wxwidgets/vcpkg_abi_info.txt
~~~

**Comment by silverqx** [comment]
> ```
> $ grep -l "^curl " installed/x64-linux/share/*/vcpkg_abi_info.txt
> ```

Thx, very useful trick. 👌

**Comment by silverqx** [comment]
The same in pwsh is:

```pwsh
gci installed/x64-windows/share/*/vcpkg_abi_info.txt | sls -Pattern '^zlib ' | Select-Object -Property Path
```
